ONSdigital/sdx-transform-cs
# Corrected vacancies transformations

## What? and Why?
Vacancies has some weird requirements.  Initially we were told that downstream wanted vacancies surveys (182-185) to be counted as 181,  this was the original version and reasonably easy to do.

Only later we were told common-software needed the PCK and the csv index file needed to be sent as 181 (as before) but IDBR needed the receipt to be of the original formtype.  We've never done this before and it wasn't really designed to be like this.  

I figured the best way to do this would to treat the submission as normal wherever possible, and the only specific bits that need modifying would be modified.  I believe this is better then doing it the opposite way (where we change it early and change it back only for the 1 bit that needs it) as it doesn't require the next person who develops on this to have any implicit knowledge about this special case, everything is as normal unless the code says otherwise (and there are comments around every bit I've had to do this to give future developers some context).

## How to test

- Spin up sdx in compose (with sdx-console and sdx-validate on the same branch name as this one)
- Put through a 182.0006-184.0006 submission (doesn't matter which)
- Check the FTP tab.  You should notice the following
  - The IDBR receipt has the original survey id in it
  - The PCK filename is 181_XXXX and has VT6A in it
  - The indicies filename begins with `EDC_181_` and there is a 181 value in it
  - The original json file is named with the original survey_id and contains the original survey id in the payload

- Do the same for 185.0005
- Check the FTP.  The only difference between this and the previous one is the PCK should contain VT5A

## Checklist
